### PR TITLE
Issue29

### PR DESCRIPTION
**Pull request by PaulinaKoltun**

### CODE REVIEW

File: src/main/java/org/pethotel/HeavenForPets/repository/RoomRepository.java [hunk 1]
```diff
@@ -33,4 +34,16 @@
...
+    List<RoomEntity> findAllPetRooms();
+
+    @Query("SELECT count(e) FROM RoomEntity e WHERE e.petType is not null")
+    int countNumbersOfPetRooms();
+
+    @Query("SELECT count(e) FROM RoomEntity e WHERE e.petType is null")
+    int countNumbersOfPlantRooms();
+
+    @Query("SELECT e FROM RoomEntity e JOIN e.shelfEntities se where se.plantInsolation = :aa and se.free = '1' and (e.temperature between :minTemp and :maxTemp)")
+    List<RoomEntity> getAllRoomsWithProperShelves(@Param ("aa") PlantInsolation plantInsolation, @Param ("minTemp") int minTemp, @Param ("maxTemp") int maxTemp);
```
Comment (robertszczygielski) [inline src/main/java/org/pethotel/HeavenForPets/repository/RoomRepository.java]: "aa" is not good name, change to "pi" or "plantInsolation"

File: src/main/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImpl.java [hunk 2]
```diff
@@ -53,19 +63,34 @@ public void saveShelves(List<Shelf> shelves, int id) {
...
+            for (Shelf shelf : shelves) {
+                if (shelf.isFree() && plantInsolation.equals(shelf.getPlantInsolation())){
+                    properShelves.add(shelf);
+                }
             }
         }
 
-        return shelves;
+
+//        RoomEntity roomEntity = roomService.findById(id);
```
Comment (robertszczygielski) [inline src/main/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImpl.java]: Remove this comment

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java [hunk 3]
```diff
@@ -245,61 +269,63 @@ public void checkIfRepositoryShowsRoomByNumber() throws Exception{
...
-        PetEntity petEntity = preparePetEntity(TEMPERATURE_20, TEMPERATURE_20);
-
-        when(petService.getPetById((long) ONE)).thenReturn(petEntity);
-        when(roomRepository.findAllPlantRooms()).thenReturn(roomEntityList);
-        when(roomMap.map(roomEntity)).thenReturn(room);
-
-        List<Room> roomList = roomService.getAllRoomsInTheRangeForPlant(ONE);
-        assertEquals(ONE, roomList.size());
-    }
+//    @Test
```
Comment (robertszczygielski) [inline src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: Why comment?

File: src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java [hunk 4]
```diff
@@ -194,7 +199,7 @@ public int getNumberOfRoomsFromQuery() {
             throw new TemperatureWrongRangeException();
         }
 
-        List<RoomEntity> roomEntityList = roomRepository.findAllPlantRooms();
+        List<RoomEntity> roomEntityList = getAllPlantRooms();
```
Comment (robertszczygielski) [inline src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java:215]: You can create in Repository method called: "getAllPlantRoomsForTemperature(min, max)" then the DB will prepare collections of Rooms with temperature and then the checking in loop is not neeaded:

 if (petEntity.getMinTemperature() <= roomEntity.getTemperature()
&& petEntity.getMaxTemperature() >= roomEntity.getTemperature()){

**Review by robertszczygielski — COMMENTED** [review]
to correct

File: src/main/java/org/pethotel/HeavenForPets/repository/RoomRepository.java (hunk 1 above)
Comment (PaulinaKoltun) [reply src/main/java/org/pethotel/HeavenForPets/repository/RoomRepository.java]: done

File: src/main/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImpl.java (hunk 2 above)
Comment (PaulinaKoltun) [reply src/main/java/org/pethotel/HeavenForPets/service/impl/ShelfServiceImpl.java]: done

File: src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java (hunk 3 above)
Comment (PaulinaKoltun) [reply src/test/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImplTest.java]: corrected

File: src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java (hunk 4 above)
Comment (PaulinaKoltun) [reply src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java:215]: Will be added in future issue

**Pull request merged**